### ISSUE TITLE
Matlab wrapper, Values::update with Point2 fails

### ISSUE THREAD

**Issue by la-sve** [borglab/gtsam#1477]
## Description
A gtsam Point2 type in Matlab can not be updated within a Values class instance. There is some conflict with dynamic and static matrix types.

## Steps to reproduce

Use GTSAM 4.2a9 with Matlab Toolbox compiled, start Matlab and add the proper paths for using the GTSAM Matlab wrapper.

### Example 1:
`a=gtsam.Values(); a.insert(0,[1;3]);a.update(0,[3;3]);`

**Results in:**
```
Error using gtsam_wrapper
Exception from gtsam:
Attempting to retrieve value with key "0", type stored in Values is
gtsam::GenericValue<Eigen::Matrix<double, -1, 1, 0, -1, 1> > but requested type was
gtsam::GenericValue<Eigen::Matrix<double, 2, 1, 0, 2, 1> >

Error in gtsam.Values/update (line 1395)
        gtsam_wrapper(2567, this, varargin{:});
```

### Example 2:
`a=gtsam.Values(); a.insert(0,gtsam.Point2(1,3));a.update(0,gtsam.Point2(3,3));`

**Results in:**
```
Error using gtsam_wrapper
Exception from gtsam:
Attempting to retrieve value with key "0", type stored in Values is
gtsam::GenericValue<Eigen::Matrix<double, -1, 1, 0, -1, 1> > but requested type was
gtsam::GenericValue<Eigen::Matrix<double, 2, 1, 0, 2, 1> >

Error in gtsam.Values/update (line 1395)
        gtsam_wrapper(2567, this, varargin{:});
```

## Expected behavior

The value at index 0 should be updated without error.

## Environment

- Ubuntu 20.04
- GTSAM 4.2a9
- Matlab R2022a

**Comment by ProfFan** [comment]
I think this is related to a similar issue in Python, we should already have checks in the C++ side that does conversion, at least for Point3...

**Comment by mattking-smith** [comment]
@ProfFan What is the difference between `a.insert()` versus `a.insert_or_assign()`? 

To me on the `MATLAB` side if I call the code with `a.insert()` land try an update with `a.update()` like @la-sve I see the same errors he is reporting. 

But if I call: 
```
a.insert_or_assign(0,[2;3])
``` 
then go for a values update with 
```
a.update(0,[1;3])
```
the `GTSAM` gods are happy and no error is thrown. What gives?

**Comment by ProfFan** [comment]
Still a type overload problem

**Comment by dellaert** [comment]
I don't use MATLAB anymore - an issue perhaps for @mattking-smith with @varunagrawal to nail down :-)

**Comment by varunagrawal** [comment]
So I figured this out. The wrapper has the overloaded `Values::insert` methods ordered as:

```cpp
void insert(size_t j, Matrix matrix);
void insert(size_t j, Vector vector);
void insert(size_t j, const gtsam::Point2& point2);
```

but the overloaded `Values::update` methods are ordered as:

```cpp
void insert(size_t j, const gtsam::Point2& point2);
void insert(size_t j, Matrix matrix);
void insert(size_t j, Vector vector);
```

When calling those methods, for `insert` the version with Vector is the first one matched, which is why the type of the stored object is `Eigen::Matrix<double, -1, 1, 0, -1, 1>`, but when calling `update`, the `Point2` version is the first match and ends up being called.

The fix is to just re-order the `insert` method so that the Point2 version is before the vector version.

**Comment by varunagrawal** [comment]
@la-sve check it out. This is the fix working on my local machine (Mac M1 no less)

<img width="747" alt="image" src="https://github.com/borglab/gtsam/assets/975964/87d3d2dd-30ff-4a0f-8c5c-0a803b972913">

**Comment by varunagrawal** [comment]
Fixed by #1654
